src/utils/BucketUtils.ts

```diff
@@ -0,0 +1,384 @@
+import { filter, flatMap, flow, forEach, includes, isEmpty, map, uniq, values } from 'lodash/fp'
+import { isNil } from 'lodash'
+import { Match } from 'src/libs/ajax/Match'
+import { DataSet } from 'src/libs/ajax/DataSet.js'
+import { processVotesForBucket } from './DarCollectionUtils'
+import { processMatchData } from './VoteUtils'
+import {
+  AlgorithmResult, DacTerm,
+  DarCollection,
+  DataAccessRequest,
+  Dataset,
+  DatasetTerm,
+  DataUseSummary,
+  DataUseTerm,
+  Election,
+  MatchResult,
+  Vote,
+} from 'src/types/model'
+import { Notifications } from 'src/libs/utils'
+import { extractError } from 'src/utils/ErrorUtils'
+import { ControlledAccessType } from 'src/libs/dataUseTranslation'
+
+export interface Bucket {
+  key: string
+  label: string
+  datasets: Dataset[]
+  datasetIds: number[]
+  dataUse?: DataUseSummary
+  dataUses?: DataUseTerm[]
+  elections: Election[]
+  votes: Record<string, VoteGroup>[]
+  matchResults: MatchResult[]
+  algorithmResult?: AlgorithmResult
+  isRP?: boolean
```

Comment:
I hadn't seen the "RP" initialism in DUOS before, and it wasn't immediately obvious what it meant.  Can a brief description be added here?

```suggestion
  isRP?: boolean // Whether this is an RP (Research Purpose) vote bucket
```